### PR TITLE
Update main fork from davepl's hexagon work

### PR DESCRIPTION
**Pull request by davepl**
Integrate all changes from the hexagon branch, including hex drawing and improved audio effects

* [X] I read the contribution guidelines in [CONTRIBUTING.md](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/CONTRIBUTING.md).
* [X] I understand the BlinkenPerBit metric, and maximized it in this PR.
* [X] I selected `main` as the target branch.
* [X] All code herein is subjected to the license terms in [COPYING.txt](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/COPYING.txt).

### CODE REVIEW

File: include/effects/matrix/spectrumeffects.h [hunk 1]
```diff
@@ -585,15 +576,131 @@ class GhostWave : public WaveformEffect
...
+        jsonDoc[PTY_SPEED]    = _scrollIncrement;
+        jsonDoc[PTY_DELTAHUE] = _hueIncrement;
+        jsonDoc[PTY_HUESTEP]  = _hueStep;
+
+        return jsonObject.set(jsonDoc.as<JsonObjectConst>());
+    }
+
+    virtual size_t DesiredFramesPerSecond() const override
+    {
+        return 90;
```
Comment (davepl) [inline include/effects/matrix/spectrumeffects.h:645]: Yes it really works and runs at 90Hz :-)

File: include/soundanalyzer.h [hunk 2]
```diff
@@ -415,7 +417,8 @@ class SoundAnalyzer : public AudioVariables
         // We'll use the average as the gVU.  I assume the average of the samples tracks sound pressure level, but don't really know...
 
         float newval = averageSum / (MAX_SAMPLES / 2 - 2);
-        debugV("AverageSumL: %f", averageSum);
+        debugV("AverageSum : %f", averageSum);
```
Comment (davepl) [inline include/soundanalyzer.h:420]: I don't think these should be Verbose, not sure how to correct once we're in a review?

Comment (davepl) [reply include/soundanalyzer.h:420]: Actually I had it backwards, I DO want these to be verbose only.  I was thinking backwards!

Comment (rbergen) [reply include/soundanalyzer.h:420]: Well in that case it's even easier: you should **not** push a change to davepl:main. :smile:

**Review by rbergen — APPROVED** [review]
LGTM! I also like the rhythm and rhyme in the title :)

**Pull request merged**